### PR TITLE
Fix func proto with qualifier in OC

### LINKED ISSUE #3766: FUNC_PROTO not correctly identified when OC block is present as param with qualifier
[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastebin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"

Similar issue to #2656, but with qualifiers. See example below:

cfg file
```
sp_arith                        = force
```

input
```
void test(NSString *param1, void (^_Nullable completionBlock)(int));
```

output
```
void test(NSString * param1, void (^_Nullable completionBlock)(int));
```

The * is being detected incorrectly as ARITH.

Will have a PR up soon

### PR DESCRIPTION
**Pull request by PoeticPete**
This PR fixes #3766 

Similar issue to #2656, but with qualifiers. See example below:

cfg file
```
sp_arith                        = force
```

input
```
void test(NSString *param1, void (^_Nullable completionBlock)(int));
```

output
```
void test(NSString * param1, void (^_Nullable completionBlock)(int));
```

The * is being detected incorrectly as ARITH.

### CODE REVIEW

**Comment by PoeticPete** [comment]
I have another PR that I want to open here: https://github.com/PoeticPete/uncrustify/pull/2

However Github doesn't automatically point PRs to the new HEAD with forked repos. Would it be possible to get write access to non-master branches?

**Comment by micheleCTDEAdmin** [comment]
Hi @PoeticPete,
would you mind squashing the commits into one? No point into committing temporary log stuff that gets removed in subsequent commits, it just adds noise :-)

Regarding write access to non-master branches, I will leave this to @guy-maurel, since he is the official maintainer. But I am not sure if that helps with the problem you mentioned. For example, if I need to add more commits to a PR, I add then to the original branch and push and the branch HEAD seems to get updated. And I avoid having PR branches on top of other PR branches, it seems to be a bit messy. I usually wait till the first PR is merged, then open a second one. Or adds the commits on the branch of the first PR alternatively.

**Review by micheleCTDEAdmin — CHANGES_REQUESTED** [review]
Please squash to avoid clutter.

**Comment by PoeticPete** [comment]
@micheleCTDEAdmin there's a way to squash and merge on github. We might need to enable it for this repo: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests

I'll do it manually for the time being

**Comment by PoeticPete** [comment]
There's a good chance I'm just missing something for forked repos.. but github does automatically switch the base branch of stacked PRs for you.

>As soon as you merge and delete the A branch, the base branch of the B pull request — which was A originally — will switch to main.
https://pspdfkit.com/blog/2021/how-to-handle-stacked-pull-requests-on-github/

**Comment by micheleCTDEAdmin** [comment]
> There's a good chance I'm just missing something for forked repos.. but github does automatically switch the base branch of stacked PRs for you.

Github does indeed update the base point of the B branch, but will show the differences between B's HEAD and the common ancestor with the main/master branch. Depending on what was done in the A branch, this may show more commits than necessary.

```
Remember that this command shows the contents of B, starting at the common ancestor between main and B. In other words, this is equivalent to git diff $(git merge-base main B) B. The problem is that the common ancestor of both branches is the commit where the A branch started diverging from main.
```

In most of my projects, I work with self-hosted gitea rather than github and we avoid merge commit on purpose. We instead rebase and fast-forward. This may seems a bit problematic (especially if commits are GPG-signed and we need to ask users to rebase them), but comes with two main benefit:
1. history is linear. It's a great thing when you often need to go back in history to track/debug stuff
2. merge commits can sometimes (not often) introduce subtle bugs because two independent PR may affect the same area and what was working in one of the PR may no longer work after merging on top of the other (yes, it happened to us). By rebasing and testing again, this can be caught before committing to the main/master branch.

**Comment by micheleCTDEAdmin** [comment]
> @micheleCTDEAdmin there's a way to squash and merge on github. We might need to enable it for this repo: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests
> 
> I'll do it manually for the time being

Yes, it is indeed possible to do it from github. But I consider it bad practice because the resulting commit will get a new hash and the user local repo may get out of sync. For example if the user has already perform a local merge to its main, their copy and the online repo would see different commit hashes (even if the changes were the same).

**Review by micheleCTDEAdmin — APPROVED** [review]
Looks good, thanks.

**Pull request merged**
